### PR TITLE
Project commit

### PR DESCRIPTION
**Pull request by RiukSan**

### CODE REVIEW

File: AccountManager/.springBeans [hunk 1]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (mx510ss) [inline AccountManager/.springBeans:1]: This file should be ignored and not commited

File: AccountManager/.gitignore [hunk 2]
```diff
@@ -0,0 +1,4 @@
+/target/
```
Comment (mx510ss) [inline AccountManager/.gitignore:1]: This file should be ignored and not commited

File: AccountManager/src/main/java/com/instinctools/trainerships/AccountsController.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+
+import com.instinctools.trainerships.services.impl.JPAAccountService;
+
+@Controller
+public class AccountsController {
+
+	ApplicationContext context = new AnnotationConfigApplicationContext(JPAAccountService.class);
+	JPAAccountService accountService = context.getBean(JPAAccountService.class);
```
Comment (mx510ss) [inline AccountManager/src/main/java/com/instinctools/trainerships/AccountsController.java]: Why not just Autowire this?

File: AccountManager/src/main/java/com/instinctools/trainerships/model/Client.java [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.JoinTable;
+import javax.persistence.OneToMany;
+import javax.persistence.Table;
+
+@Entity
+@Table
+public class Client implements Serializable {
+
```
Comment (mx510ss) [inline AccountManager/src/main/java/com/instinctools/trainerships/model/Client.java]: In Account you set names for table and all fields - but for Client not
Lets follow some unified convention - better to add table and column names here also

File: AccountManager/src/main/java/com/instinctools/trainerships/services/impl/JPAAccountService.java [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+import org.hibernate.Transaction;
+import org.springframework.stereotype.Service;
+
+import com.instinctools.trainerships.HibernateUtil;
+import com.instinctools.trainerships.model.Account;
+import com.instinctools.trainerships.model.Client;
+import com.instinctools.trainerships.services.AccountService;
+
+@Service
+public class JPAAccountService implements AccountService {
```
Comment (mx510ss) [inline AccountManager/src/main/java/com/instinctools/trainerships/services/impl/JPAAccountService.java]: Based on what you doing in this "Service"  I would say that it not a service at all - it's a data access layer which works with only JPA/SQL. One of the next tasks will be to add support of NoSQL storage - and it should be done like implement some new Data access layer and just inject it to existing real services - In your case its impossible to complete because your service is hard bounded to SQL storage
So to summarize - you need one more layer - Data access layer with it own interfaces and etc.

File: AccountManager/src/main/java/com/instinctools/trainerships/services/impl/JPAClientService.java [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+import org.springframework.stereotype.Service;
+
+import com.instinctools.trainerships.HibernateUtil;
+import com.instinctools.trainerships.model.Client;
+import com.instinctools.trainerships.services.ClientService;
+
+@Service
+public class JPAClientService implements ClientService {
```
Comment (mx510ss) [inline AccountManager/src/main/java/com/instinctools/trainerships/services/impl/JPAClientService.java]: The same as for JPAAccountService

**Review by mx510ss — CHANGES_REQUESTED** [review]
Need Data Access Layer between Service layer and hibernate things

File: AccountManager/pom.xml [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+	<properties>
+		<java-version>1.6</java-version>
+		<org.springframework-version>4.3.11.RELEASE</org.springframework-version>
+		<org.aspectj-version>1.6.10</org.aspectj-version>
+		<org.slf4j-version>1.6.6</org.slf4j-version>
+	</properties>
+	<dependencies>
+		<!-- Spring -->
+		<dependency>
+			<groupId>org.springframework</groupId>
```
Comment (mx510ss) [inline AccountManager/pom.xml:19]: Actually spring will be in the future and in this task you shouldn't use it
but okay - let it be as is

File: AccountManager/pom.xml [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+			</exclusions>
+		</dependency>
+		<dependency>
+			<groupId>org.springframework</groupId>
+			<artifactId>spring-webmvc</artifactId>
+			<version>${org.springframework-version}</version>
+		</dependency>
+		<!-- Logging -->
+		<dependency>
+			<groupId>org.slf4j</groupId>
```
Comment (mx510ss) [inline AccountManager/pom.xml:37]: Half of your dependencies - is a logging related
Don't think we need them all - just add org.slf4j + logback and no need all others

Comment (mx510ss) [reply AccountManager/pom.xml:37]: Or you can use spring defaults logging

File: AccountManager/pom.xml [hunk 9]
```diff
@@ -0,0 +1,136 @@
...
+		<dependency>
+			<groupId>org.hibernate</groupId>
+			<artifactId>hibernate-validator</artifactId>
+			<version>5.2.4.Final</version>
+		</dependency>     
+	</dependencies>
+    <build>
+        <plugins>
+            <plugin>
+                <artifactId>maven-eclipse-plugin</artifactId>
```
Comment (mx510ss) [inline AccountManager/pom.xml:101]: Some strange IDE specific plugin. How it helps you?

File: AccountManager/pom.xml [hunk 10]
```diff
@@ -0,0 +1,136 @@
...
+                    <downloadSources>true</downloadSources>
+                    <downloadJavadocs>true</downloadJavadocs>
+                </configuration>
+            </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <version>2.5.1</version>
+                <configuration>
+                    <source>1.6</source>
```
Comment (mx510ss) [inline AccountManager/pom.xml:119]: 1.6 is obsolete. Please switch at least to 1.8

File: AccountManager/src/main/java/com/instinctools/trainerships/AccountsController.java [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+
+import com.instinctools.trainerships.services.impl.AccountServiceImpl;
+
+@Controller
+public class AccountsController {
+
+	ApplicationContext context = new AnnotationConfigApplicationContext(AccountServiceImpl.class);
+	AccountServiceImpl accountService = context.getBean(AccountServiceImpl.class);
+	
+	@RequestMapping(value = "/updateAccount", method = RequestMethod.GET)
```
Comment (mx510ss) [inline AccountManager/src/main/java/com/instinctools/trainerships/AccountsController.java:21]: By naming convention we use GET method only to retrieve data from API
so to create something u should use POST
to update PUT
to DELETE - DELETE
so please make changes in all places, where is needed

File: AccountManager/src/main/java/com/instinctools/trainerships/ClientController.java [hunk 12]
```diff
@@ -0,0 +1,49 @@
...
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+
+import com.instinctools.trainerships.model.Client;
+import com.instinctools.trainerships.services.impl.ClientServiceImpl;
+
+@Controller
+public class ClientController {
+
+	ApplicationContext context = new AnnotationConfigApplicationContext(ClientServiceImpl.class);
```
Comment (mx510ss) [inline AccountManager/src/main/java/com/instinctools/trainerships/ClientController.java:20]: if you using spring - just use @Autowired to get beans

File: AccountManager/src/main/java/com/instinctools/trainerships/model/Account.java [hunk 13]
```diff
@@ -0,0 +1,55 @@
...
+import javax.persistence.JoinColumn;
+import javax.persistence.JoinTable;
+import javax.persistence.ManyToOne;
+import javax.persistence.Table;
+import javax.persistence.UniqueConstraint;
+
+@Entity 
+@Table(name = "Account", uniqueConstraints = { 
+        @UniqueConstraint(columnNames = "ID")}) 
+public class Account implements Serializable {
```
Comment (mx510ss) [inline AccountManager/src/main/java/com/instinctools/trainerships/model/Account.java:20]: its not require to implements Serializable at this point

File: AccountManager/src/main/java/com/instinctools/trainerships/model/Client.java [hunk 14]
```diff
@@ -0,0 +1,69 @@
...
+import javax.persistence.JoinColumn;
+import javax.persistence.JoinTable;
+import javax.persistence.OneToMany;
+import javax.persistence.Table;
+import javax.persistence.UniqueConstraint;
+
+@Entity
+@Table(name = "Client", uniqueConstraints = { 
+        @UniqueConstraint(columnNames = "ID")}) 
+public class Client implements Serializable {
```
Comment (mx510ss) [inline AccountManager/src/main/java/com/instinctools/trainerships/model/Client.java:22]: the same about serializable - its not needed here

File: AccountManager/src/main/webapp/WEB-INF/spring/appServlet/servlet-context.xml [hunk 15]
```diff
@@ -0,0 +1,28 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (mx510ss) [inline AccountManager/src/main/webapp/WEB-INF/spring/appServlet/servlet-context.xml:1]: Try to not use Xml-based spring configuration - replace by annotations

File: AccountManager/src/test/resources/log4j.xml [hunk 16]
```diff
@@ -0,0 +1,41 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE log4j:configuration PUBLIC "-//APACHE//DTD LOG4J 1.2//EN" "log4j.dtd">
+<log4j:configuration xmlns:log4j="http://jakarta.apache.org/log4j/">
+
```
Comment (mx510ss) [inline AccountManager/src/test/resources/log4j.xml:4]: you already have log4j.xml
why you need another one - is there some specific loggers for tests only?
